### PR TITLE
Remove roadrunner specific session handling

### PR DESCRIPTION
**Pull request by alexander-schranz**
This is build on top of the following refractoring: https://github.com/symfony/symfony/pull/41390

Fixes:

 - [x] Problem E at https://github.com/php-runtime/runtime/issues/46
 - [x] Problem F at https://github.com/php-runtime/runtime/issues/46

TODO:

 - [x] Requires merge: https://github.com/symfony/symfony/pull/41390
 - [x] Requires composer.json update

### CODE REVIEW

File: src/roadrunner-symfony-nyholm/src/Runner.php [hunk 1]
```diff
@@ -73,6 +52,7 @@ public function run(): int
             } catch (\Throwable $e) {
                 $worker->getWorker()->error((string) $e);
             } finally {
+                // TODO move this cleanup also to Symfony
```
Comment (alexander-schranz) [inline src/roadrunner-symfony-nyholm/src/Runner.php:55]: The following part is also required in all cases, but not yet sure where to put it into symfony?

```php
                if (PHP_SESSION_ACTIVE === session_status()) {
                    session_abort();
                }
                // reset all session variables to initialize state
                $_SESSION = [];
session_id(''); // in this case session_start() will generate us a new session_id()
```

**Comment by Nyholm** [comment]
Can we update this PR now when https://github.com/symfony/symfony/pull/41390 is merged?

**Comment by alexander-schranz** [comment]
@Nyholm rebased and updated the dependencies.

File: src/roadrunner-symfony-nyholm/src/Runner.php [hunk 2]
```diff
@@ -27,33 +22,12 @@ class Runner implements RunnerInterface
...
-     */
-    public function __construct($kernel, ?HttpFoundationFactoryInterface $httpFoundationFactory = null, ?HttpMessageFactoryInterface $httpMessageFactory = null)
+    public function __construct(HttpKernelInterface $kernel, ?HttpFoundationFactoryInterface $httpFoundationFactory = null, ?HttpMessageFactoryInterface $httpMessageFactory = null)
     {
         $this->kernel = $kernel;
         $this->psrFactory = new Psr7\Factory\Psr17Factory();
         $this->httpFoundationFactory = $httpFoundationFactory ?? new HttpFoundationFactory();
         $this->httpMessageFactory = $httpMessageFactory ?? new PsrHttpFactory($this->psrFactory, $this->psrFactory, $this->psrFactory, $this->psrFactory);
-
-        if ($kernel instanceof HttpCache) {
```
Comment (Nyholm) [inline src/roadrunner-symfony-nyholm/src/Runner.php:45]: Why isn’t this need anymore?

**Review by Nyholm — COMMENTED** [review]
Awesome. Thank you. 

Why was the HttpCache support removed?

File: src/roadrunner-symfony-nyholm/src/Runner.php (hunk 2 above)
Comment (alexander-schranz) [reply src/roadrunner-symfony-nyholm/src/Runner.php:45]: ` if ($kernel instanceof HttpCache) {` was only needed to get the paramters from the wrapped kernel. As we are not longer need to handle the sessions here we could remove this. The support for `HttpCache` is still here as `HttpCache` implements `HttpKernelInterface`. Did update the tests which create the Runner instances.

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**